packages/sdk/src/instant-trade/InstantTradeBuyerTxBuilder.ts

```diff
@@ -0,0 +1,199 @@
+import { Psbt } from "bitcoinjs-lib"
+import reverseBuffer from "buffer-reverse"
+
+import {
+  decodePSBT,
+  generateTxUniqueIdentifier,
+  getScriptType,
+  INSTANT_BUY_SELLER_INPUT_INDEX,
+  OrditApi,
+  processInput
+} from ".."
+import { MINIMUM_AMOUNT_IN_SATS } from "../constants"
+import { InjectableInput, InjectableOutput } from "../transactions/PSBTBuilder"
+import { Output } from "../transactions/types"
+import InstantTradeBuilder, { InstantTradeBuilderArgOptions } from "./InstantTradeBuilder"
+
+interface InstantTradeBuyerTxBuilderArgOptions extends InstantTradeBuilderArgOptions {
+  feeRate: number
+  sellerPSBT: string
+  receiveAddress?: string
+}
+
+export default class InstantTradeBuyerTxBuilder extends InstantTradeBuilder {
+  receiveAddress?: string
+  sellerPSBT!: Psbt
+  sellerAddress?: string
+
+  constructor({
+    address,
+    network,
+    publicKey,
+    receiveAddress,
+    sellerPSBT,
+    feeRate
+  }: InstantTradeBuyerTxBuilderArgOptions) {
+    super({
+      address,
+      network,
+      publicKey
+    })
+
+    this.feeRate = feeRate
+    this.receiveAddress = receiveAddress
+    this.decodeSellerPSBT(sellerPSBT)
+  }
+
+  private decodeSellerPSBT(hex: string) {
+    this.sellerPSBT = decodePSBT({ hex })
+    this.inscriptionOutpoint = generateTxUniqueIdentifier(
+      reverseBuffer(this.sellerPSBT.txInputs[0].hash).toString("hex"),
+      this.sellerPSBT.txInputs[0].index
+    )
+
+    const [input] = this.sellerPSBT.data.inputs
+    if (!input?.witnessUtxo) {
+      throw new Error("invalid seller psbt")
+    }
+
+    const data = getScriptType(input.witnessUtxo.script, this.network)
+    this.sellerAddress = data.payload && data.payload.address ? data.payload.address : undefined
+    if (!this.sellerAddress) {
+      throw new Error("invalid seller psbt")
+    }
+  }
+
+  private decodePrice() {
+    this.validatePrice((this.sellerPSBT.data.globalMap.unsignedTx as any).tx.outs[0].value - this.postage)
```

Comment:
I am not sure if validating price (minus postage) to be less than 600 (min) makes sense. Because when buyer will send `sellingprice + postage` to seller as one output. so as long as `sellingprice + postage` > 600 it will always be valid?